src/multivariate/mvdiscretenonparametric.jl

```diff
@@ -0,0 +1,153 @@
+        new{T,P,Ts,Ps}(support, p)
+    end
+end
+
+"""
+    MvDiscreteNonParametric(
+        support::AbstractVector,
+        p::AbstractVector{<:Real}=fill(inv(length(support)), length(support)),
+    )
+
+Construct a multivariate discrete nonparametric probability distribution with `support` and corresponding
+probabilities `p`. If the probability vector argument is not passed, then
+equal probability is assigned to each entry in the support.
+
+# Examples
+```julia
+using ArraysOfArrays
+# rows correspond to samples
+μ = MvDiscreteNonParametric(nestedview(rand(7,3)'))
+
+# columns correspond to samples
+ν = MvDiscreteNonParametric(nestedview(rand(7,3)))
+```
+"""
+function MvDiscreteNonParametric(
+    support::AbstractVector{<:AbstractVector{<:Real}},
+    p::AbstractVector{<:Real}=fill(inv(length(support)), length(support)),
+)
+    return MvDiscreteNonParametric{eltype(eltype(support)),eltype(p),typeof(ArrayOfSimilarArrays(support)),typeof(p)}(
+        ArrayOfSimilarArrays(support), p)
+end
+
+"""
+    MvDiscreteNonParametric(
+        support::Matrix{<:Real},
+        p::AbstractVector{<:Real}=fill(inv(length(support)), length(support)
+    )
+
+Construct a multivariate discrete nonparametric probability distribution
+from a matrix as `support` where each row is a sample, and corresponding
+probabilities `p`. If the probability vector argument is not passed, then
+equal probability is assigned to each entry in the support.
+
+# Examples
+```julia
+# the rows correspond to the samples
+using LinearAlgebra
+μ = MvDiscreteNonParametric(rand(10,3), normalize!(rand(10),1))
+```
+"""
+function MvDiscreteNonParametric(
+    support::Matrix{<:Real},
+    p::AbstractVector{<:Real}=fill(inv(size(support)[1]), size(support)[1])
+)
+    return MvDiscreteNonParametric(nestedview(support'), p)
+end
+
+Base.eltype(::Type{<:MvDiscreteNonParametric{T}}) where T = T
+
+"""
+    support(d::MvDiscreteNonParametric)
+Get a sorted AbstractVector defining the support of `d`.
+"""
+support(d::MvDiscreteNonParametric) = d.support
+
+"""
+    probs(d::MvDiscreteNonParametric)
+Get the vector of probabilities associated with the support of `d`.
+"""
+probs(d::MvDiscreteNonParametric) = d.p
+
+
+# It would be more intuitive if length was the
+# 
+"""
+    length(d::MvDiscreteNonParametric)
+Retunrs the dimension of the mass points (samples).
+It corresponds to `innersize(d.support)[1]`,
+where `innersize` is a function from from ArraysOfArrays.jl.
+"""
+Base.length(d::MvDiscreteNonParametric) = innersize(d.support)[1]
+
+"""
+    size(d::MvDiscreteNonParametric)
+Returns the size of the support as a tuple where
+the first value is the number of points (samples)
+and the second is the dimension of the samples (e.g ℝⁿ).
+It corresponds to `size(flatview(d.support)')`
+where `flatview` is a function from from ArraysOfArrays.jl
+that turns an Array of Arrays into a matrix.
+"""
+Base.size(d::MvDiscreteNonParametric) = size(flatview(d.support)')
```

Comment:
For instance, here one could just use
```suggestion
Base.length(d::MvDiscreteNonParametric) = length(first(d.support))
Base.size(d::MvDiscreteNonParametric) = (length(d), length(d.support))
```